RoaringBitmap/src/main/java/org/roaringbitmap/RoaringBitmap.java

```diff
@@ -1338,248 +1338,78 @@ public static RoaringBitmap andNot(final RoaringBitmap x1, final RoaringBitmap x
-      Container c2 = Container.rangeOfOnes(0, containerLast + 1);
-
-      if (s1 == s2) {
-        final Container c1 = src.highLowContainer.getContainerAtIndex(pos1);
-        // If we re not at the last container, just use the full container.
-        // Otherwise, compute an in-place or.
-        final Container c = (s2 == lastKey) ? (inplace ? c1.ior(c2) : c1.or(c2)) : c2;
-        if (destPos < dest.highLowContainer.size()) {
-          dest.highLowContainer.replaceKeyAndContainerAtIndex(destPos, s1, c);
-        } else {
-          dest.highLowContainer.insertNewKeyValueAt(destPos, s1, c);
-        }
-
-        pos1++;
-        s2++;
-        destPos++;
-      } else if (s1 - s2 > 0) { 
-        dest.highLowContainer.insertNewKeyValueAt(destPos, s2, c2);
-        pos1 += insertionIncrement;
-        length1 += insertionIncrement;
-        s2++;
-        destPos++;
-      } else { 
-        throw new IllegalStateException("This is a bug. Please report to github");
-      }
-    }
-    return s2;
-  }
-
-  /*
-   * Handle the orNot operations for the remaining containers in the other Bitmap.
-   * Two cases here:
-   * 1. either we have a hole. In this case, a full container should be appended.
-   * 2. or we have a container. an inplace orNot is applied and the result is appended.
-   *
-   * Note that, at this stage, all the own containers were treated.
-   * That's why we only have to append.
-   */
-  private static char orNotHandleRemainingOtherContainers(
-          final RoaringBitmap other, final RoaringBitmap dest, int pos2,
-          int length2, char s2, char lastKey, int lastSize) {
-    while (pos2 < length2 && s2 - lastKey <= 0) { // s2 <= lastKey
-      final int containerLast = (s2 == lastKey) ? lastSize : Util.maxLowBitAsInteger();
-      if (s2 == other.highLowContainer.getKeyAtIndex(pos2)) {
-        final Container c2 = other.highLowContainer.getContainerAtIndex(pos2);
-        Container c = new RunContainer().iorNot(c2, containerLast + 1);
-        dest.highLowContainer.append(s2, c);
-        pos2++;
-      } else {
-        dest.highLowContainer.append(s2, Container.rangeOfOnes(0, containerLast + 1));
-      }
-      s2++;
-    }
-    return s2;
-  }
-
-  /*
-   * Handle the remaining holes.
-   * A full container should be appended for each key.
-   */
-  private static void orNotHandleRemainingHoles(
-          RoaringBitmap dest, char s2, char lastKey, int lastSize) {
-    while (s2 < lastKey) { 
-      dest.highLowContainer.append(s2, RunContainer.full());
-      s2++;
-    }
-    if (s2 == lastKey) {
-      dest.highLowContainer.append(s2, Container.rangeOfOnes(0, lastSize + 1));
-    }
-  }
-
   /**
    * In-place bitwise ORNOT operation. The current bitmap is modified.
    *
    * @param other the other bitmap
-   * @param rangeEnd end point of the range (exclusive)
+   * @param rangeEnd end point of the range (exclusive); values beyond this range in
```

Comment:
There may be be an extra "in" in this sentence.

Reply:
yes, removed this now (semantics changed)